api/controllers/console/app/wraps.py

```diff
@@ -26,6 +26,31 @@ def decorated_view(*args, **kwargs):
                 .first()
             )
 
+            if not app_model:
+                # If app not found in current tenant, check other workspaces
+                accessible_tenants = (
+                    db.session.query(Tenant)
+                    .join(TenantAccountJoin, TenantAccountJoin.tenant_id == Tenant.id)
+                    .filter(TenantAccountJoin.account_id == current_user.id)
+                    .all()
+                )
```

Comment:
It's not the best way to check the user's accessiblilty of  the targeted tenant by loading all the tenants in to memory in one time. Try to use subquery or join to ensure that and minimize the momory footprints.

Reply:
ok, new commit enhance this q

Reply:
Thx. It looks much better now in looking-up the targeting tenant.